### DISCUSSION TITLE
Game Design Idea

### DISCUSSION THREAD

**Discussion by Quit975** [Quit975/DuckyGame#28] [Ideas]
Currently, we have a very crude and basic prototype for a small game based around relying on audio location to progress. We will first focus on improving on this prototype slightly, to see if this is indeed a fun gameloop. But it's not a complete game! Here's my idea for the final game that we will be working towards (obviously it's subject to change as we iterate over the prototype, and also all feedback is welcome)

So the main selling point is the "audiolocation" based gameplay - we need to listen for the invisible frog to catch it as many times as we can. But the source of the danger is visually based (enemies on the screen), so this (hopefully :o) will create the nice counter-play of "I need to listen to win, but watch not to lose" - once we iron out the prototype we will see if this is indeed fun.

But to make it an actual game, I suggest following design direction:
- Lives: the player will have a few lives (maybe classic hearts), once he loses them all, game over
- The goal is to capture the frog X times (we have a slightly more attractive counter, maybe an image of a frog and a number in the corner)
- Every time a frog is caught, everything is becoming more difficult (enemies and frog are beciming faster and faster). This could be also communicated with background music starting slow, and becoming faster and faster with each frog
- Once the player catches X frogs, music changes, enemies disappear and Pepe the Boss enters
- Boss fight should also be based on the "audiolocation vs visual" feature. Maybe in order to defeat him player needs to audiolocate something, or whatever - we can brainstorm about it at a later stage
- Defeating Pepe the Boss wins the game
- Maybe we can have a scoreboard based on how much time it took to beat the game

**Comment by pyskaa0506** [comment]
I love the idea of pepe the boss
It's funny and makes the game much more appealing. Getting to the final enemy would be something i would tryhard to get to, just to meet The Mighty Pepe.

It also would be amusing if pepe after getting some damage would make noises, like "Reeeeeee" for example.

**Comment by simplyNoOne** [comment]
I have an alternative game mode that could be added, so that the player might choose between two major modes: Standard(your idea with boss fight after catching the frog X times) and Timer.

Timer would work as follows:
Enemies and Frog behave the same way as in Standard, and there, too, is a set number of times the Frog must be caught (X) BUT the time to catch the Frog is also fixed. Namely: Players gets a random amount of time in range, say, 2-3 minutes and during that time he needs to catch the Frog X times. It would require more time management skills, as once Player reaches X Frogs caught, the game doesn't end, from then on, he would need to avoid colliding with the Frog. 
Eg. I have 2.5 minutes to catch the Frog 12 times. I manage to do it in only 2 minutes, so I need to avoid colliding with the Frog for the next 30 seconds, so that once the timer stops, I still have precisely 12 Frogs caught.

Just and idea if we wanted to further develop the game.  It is ofc a topic open for all kinds of discussion, suggestions, changes

**Reply by Quit975** [reply]
hmmmm not sure... on paper it sounds like it could be annoying for the player (having to dodge both visual entities and also invisible frog)? In any case, we definitely don't want to think about alternative modes for the time being. However, should we have some sort of random pickups during the game, some of them could be harmful for the player, and having to avoid the frog for some seconds could be a fun addition 🤔 to consider after reaching the first milestone for sure
